SimTracker/TrackerHitAssociation/plugins/ClusterTPAssociationProducer.cc

```diff
@@ -29,6 +29,17 @@
 #include "SimDataFormats/TrackingAnalysis/interface/TrackingParticleFwd.h"
 #include "SimTracker/TrackerHitAssociation/interface/ClusterTPAssociation.h"
 
+namespace {
```

Comment:
@VinInn , sorry I may be miss something but the same definition is in the header and here in the source.

Reply:
which definition?

Reply:
I did not modify any header

Reply:
I am sorry, the correct question: the same code included both inside SimHitTPAssociationProducer.cc and ClusterTPAssociationProducer.cc

namespace {
  using TkId = std::pair<uint32_t, EncodedEventId>;
  struct TkIdHash {
    std::size_t operator()(TkId const& s) const noexcept {
      std::size_t h1 = std::hash<uint32_t>{}(s.first);
      std::size_t h2 = std::hash<uint32_t>{}(s.second.rawId());
      return h1 ^ (h2 << 1);
    }
  };
}  // namespace

I understand , that it is a correct and the most performant variant but Is it unavoidable to repeat these lines in each producer? For curiosity, how much you win?

Reply:
yes, can be factorized somewhere. (Where?)
What I win is that I will not need to recompile the whole universe (if I include it in EncodedEventId for instance)  and I do not pollute the global namespace.
Actually I do not know where to locate it .... Any suggestion? One has to avoid additional dependencies (and of course the name shall be changed to be more unique etc)

Reply:
@makortel , can you comment. In a mean time I sign the PR.

Reply:
Given the dependence chain (`SimTrack::trackId()`, `EncodedEventId`), a new header in `SimDataFormats/Track/interface` could be somewhat natural place.

Reply:
Any suggestion for the name of the alias of the pair and for the hash struct (and the file itself)?
I hope that alias and struct can be located in the same file....

Reply:
Just thinking out loud , maybe something along `SimTrackAndEventId`, `SimTrackAndEncodedEventId`, `SimTrackEventIdPair`  (all with `...Hash`)? 

I'm not super happy with any of them though (and more often than not I find good names difficult to come up).

> I hope that alias and struct can be located in the same file....

Given the usage I don't see strong motivations to use different files.

Reply:
What about: UniqueSimTrackId (or SimTrackUniqueId) ?

Reply:
> What about: UniqueSimTrackId (or SimTrackUniqueId) ?

I would be fine with either, `UniqueSimTrackId` sounds maybe a bit better.